services/user/CommonSys/common-lib/packages/wasm-loader/src/main.ts

```diff
@@ -198,25 +160,37 @@ const parseFunctions = (data: string): ServiceMethodIndex => {
     return entries;
 };
 
+interface FunctionResult<T = any> {
+    method: string;
+    result: T;
+}
+
 const serviceMethodIndexToImportables = (
     serviceMethodIndex: ServiceMethodIndex,
     service: string,
-    id: string
-): { [key: string]: string }[] => {
-    return Object.entries(serviceMethodIndex).map(([key, functions]) => {
-        return {
-            [`component:${service}}/${key}`]: functions
-                .map(
-                    (func) =>
-                        generatePendingFunction(
-                            { method: func, service: key },
-                            id
-                        ) + "\n"
-                )
-                .join("\n")
-        };
-    });
-};
+    id: string,
+    functionsResult: FunctionResult[]
+): { [key: string]: string }[] =>
+    Object.entries(serviceMethodIndex).map(([key, methodNames]) => ({
+        [`component:${service}/${key}`]: methodNames
```

Comment:
The component-parser component I made was "psibase:component-parser".
An example wasi component looks like:  "wasi:io". And it's possible to only import parts of a component by using slash notation, such as `wasi:io/error` if I only want the `error` object out of the `wasi:io` component.

In WIT, they refer to these fields in the following way:
`namespace:package@semver/name`

To generalize,
* `namespace` = service
* `package` = which plugin
* `@semver` is optional, possibly forbidden for psibase packages
* `/name` = name of the item being imported from that component. If left out, the entire component is imported.